src/adapter/src/catalog/builtin.rs

```diff
@@ -137,6 +137,7 @@ pub struct BuiltinRole {
     pub name: &'static str,
 }
 
+/// Uniquely identifies the definition of a builtin object.
```

Comment:
What happens if a two different objects have the same fingerprint? Do fingerprints have to be unique among all other fingerprints, or just ones of the same object type? The word "uniquely" here conflicts with the various implementations that sometimes all produce 0, and also doesn't work with Hashable which may collide.

Reply:
Yeah this comment is incorrect. Fingerprints are never compared across objects. The only thing we care about is that when an object definition is changed, then the fingerprint changes.

Reply:
Why is it ok to use hashed for that?

Reply:
Maybe I'm misunderstanding something, but if we change the object definition, then it's hash will also change.

Reply:
Maybe we should just be serializing the definition and comparing those.

Reply:
I think it's hash *probably* changes, not will for sure change.

Reply:
hmm, yeah. I think a serialized json of the definition is probably what we want here. At least this is an existing issue so it should be OK to include this in the release.